### PR TITLE
chore(ICP-Rosetta): FI-1557: get neuron info function

### PR DESCRIPTION
**Pull request by NikolasHai**
This MR proposes the following changes:

1. Add neuron info to the icp rosetta client
2. Remove the legacy test for neuron info

### CODE REVIEW

File: rs/rosetta-api/icp/src/ledger_client/neuron_response.rs [hunk 1]
```diff
@@ -28,3 +28,16 @@ impl TryFrom<NeuronResponse> for ObjectMap {
...
+impl TryFrom<ObjectMap> for NeuronResponse {
+    type Error = ApiError;
+    fn try_from(o: ObjectMap) -> Result<NeuronResponse, Self::Error> {
+        match serde_json::from_value(Value::Object(o)) {
+            Ok(d) => Ok(d),
+            Err(err) => Err(ApiError::internal_error(format!(
+                "Could not convert ObjectMap to NeuronResponse: {:?}",
+                err
+            ))),
+        }
```
Comment (mbjorkqvist) [inline rs/rosetta-api/icp/src/ledger_client/neuron_response.rs:41]:
```suggestion
        serde_json::from_value(Value::Object(o)).map_err(|err| {
            ApiError::internal_error(format!(
                "Could not convert ObjectMap to NeuronResponse: {:?}",
                err
            ))
        })
```

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs [hunk 2]
```diff
@@ -723,3 +723,96 @@ fn test_list_known_neurons() {
...
+            .first()
+            .unwrap()
+            .clone()
+            .metadata
+            .unwrap(),
+        )
+        .unwrap();
+
+        assert_eq!(neuron_info.neuron_id, neuron.id.unwrap().id);
+        assert_eq!(neuron_info.controller.0, TEST_IDENTITY.sender().unwrap());
```
Comment (mbjorkqvist) [inline rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs:789]: Is there a reason for not verifying the other fields here, as was done in the old test?

Comment (NikolasHai) [reply rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs:789]: I am testing that the correct neuron was fetched (which is the ID). I am not testing the content of the neuron as that is something that is tested on the governance canister level.

File: rs/rosetta-api/icp/client/src/lib.rs [hunk 3]
```diff
@@ -1250,3 +1315,49 @@ impl RosettaDisburseNeuronArgsBuilder {
...
+    }
+}
+
+pub struct RosettaNeuronInfoArgsBuilder {
+    neuron_index: u64,
+    public_key: Option<PublicKey>,
+    principal_id: Option<PrincipalId>,
+}
+
+impl RosettaNeuronInfoArgsBuilder {
```
Comment (fsodre) [inline rs/rosetta-api/icp/client/src/lib.rs:1337]: Just a curiosity, not a blocking comment.

I see a few places where we manually implement the Builder pattern. It seems like there are a few crates that provide automatic creation for them (like https://github.com/elastio/bon). Have we considered using them instead?

File: rs/rosetta-api/icp/src/ledger_client.rs [hunk 4]
```diff
@@ -137,6 +138,18 @@ pub enum OperationOutput {
     ListNeuronsResponse(ListNeuronsResponse),
 }
 
+impl TryFrom<ObjectMap> for OperationOutput {
+    type Error = ApiError;
+    fn try_from(o: ObjectMap) -> Result<Self, Self::Error> {
```
Comment (fsodre) [inline rs/rosetta-api/icp/src/ledger_client.rs:143]: No issue here, just for my own learning: why do we use `Self` here but in neuron_response.rs:34 we use `NeuronResponse` specifically?

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs [hunk 5]
```diff
@@ -723,3 +723,96 @@ fn test_list_known_neurons() {
...
+    });
```
Comment (fsodre) [inline rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs:817]: Should we add a test for trying to get a neuron that doesn't exist?

File: rs/rosetta-api/icp/client/src/lib.rs (hunk 3 above)
Comment (NikolasHai) [reply rs/rosetta-api/icp/client/src/lib.rs:1337]: We usually implement the building pattern ourselves in the entire repo. Generally speaking, in the monorepo new crates try to be avoided as it is another dependency to monitor and manage.

File: rs/rosetta-api/icp/src/ledger_client.rs (hunk 4 above)
Comment (NikolasHai) [reply rs/rosetta-api/icp/src/ledger_client.rs:143]: It should not make any difference.

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs (hunk 5 above)
Comment (NikolasHai) [reply rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs:830]: Done

**Review by mbjorkqvist — APPROVED** [review]

**Pull request merged**